tests/unit/test_provider.py

```diff
@@ -120,20 +119,19 @@ def test_relation_data(self):
         "charms.loki_k8s.v0.loki_push_api.LokiPushApiProvider._remove_alert_rules_files",
         MagicMock(),
     )
-    @patch(
-        "charms.loki_k8s.v0.loki_push_api.LokiPushApiProvider.unit_ip", new_callable=PropertyMock
-    )
-    def test__on_logging_relation_changed(self, mock_unit_ip):
+    def test__on_logging_relation_changed(self):
         with self.assertLogs(level="DEBUG") as logger:
-            mock_unit_ip.return_value = "10.1.2.3"
             rel_id = self.harness.add_relation("logging", "promtail")
             self.harness.add_relation_unit(rel_id, "promtail/0")
-
             self.harness.update_relation_data(rel_id, "promtail", {"alert_rules": "ww"})
             self.assertEqual(
                 sorted(logger.output)[0],
                 "DEBUG:charms.loki_k8s.v0.loki_push_api:Saved alerts rules to disk",
             )
+            self.assertEqual(
```

Comment:
Can we assert against the relation data directly?

Reply:
Even if we can, ensuring that the actual code sees it is probably of value.